### PR TITLE
TNT-39261 [ODD] Implement context providers + geo lookup

### PR DESCRIPTION
**Pull request by XDex**
## Description

TNT-39261 [ODD] Implement context provider
TNT-39266  [ODD] Implement geo provider

## Related Issue

https://jira.corp.adobe.com/browse/TNT-39261
https://jira.corp.adobe.com/browse/TNT-39266

### CODE REVIEW

File: Source/Adobe.Target.Client/Util/MurmurHash3.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    {
+        private const int Seed = 0;
+        private const int CharBytes = 2;
+        private const uint C1 = 0xcc9e2d51;
+        private const uint C2 = 0x1b873593;
+        private const int R1 = 15;
+        private const int R2 = 13;
+        private const int M = 5;
+        private const uint N = 0xe6546b64;
+
```
Comment (artur-ciocanu) [inline Source/Adobe.Target.Client/Util/MurmurHash3.cs:23]: Please rename to `HashUnencodedChars()`

**Review by artur-ciocanu — CHANGES_REQUESTED** [review]
@XDex overall looks good, but there is no a single test. The functionality that you have added for collators and cookier parsing etc, could be tested via unit tests, so please add the required tests.

**Comment by XDex** [comment]
@artur-ciocanu Please take another look when you have a moment.. Thx!

**Review by artur-ciocanu — APPROVED** [review]
@XDex it looks good. However I am not really fond of all the code related to domain parsing that doesn't have a single unit test. The code is not trivial, so I would expect tests that over all the domain parsing and public suffix list related logic.

**Pull request merged**